File: src/main/java/my/maryvk/maryvkweb/service/RelationService.java
package my.maryvk.maryvkweb.service;

import my.maryvk.maryvkweb.domain.Relation;
import my.maryvk.maryvkweb.domain.RelationType;

import java.util.List;

public interface RelationService {
    List<Integer> findAllFriends(Integer userId);
    List<Integer> findAllFollowers(Integer userId);
    List<Integer> findAllFor(Integer userId, RelationType relationType);

    void addRelation(Relation relation);
    void removeRelation(Relation relation);
}

File: src/main/java/my/maryvk/maryvkweb/service/RelationService.kt
package my.maryvk.maryvkweb.service

import my.maryvk.maryvkweb.domain.Relation
import my.maryvk.maryvkweb.domain.RelationType

interface RelationService {
    fun findAllFor(userId: Int, relationType: RelationType): List<Int>

    fun addRelation(relation: Relation)
    fun removeRelation(relation: Relation)
}